influence_benchmark/stats/retroactive_evals.py

```diff
@@ -203,104 +197,122 @@ def save_results(self, results_df):
             iteration_df.to_json(output_path, orient="records")
             print(f"Results for iteration {iteration_number} saved to: {output_path}")
 
-    async def _async_evaluate_iteration(self, all_transcripts_with_env, total_transcripts):
+    async def _async_evaluate_iteration(self, all_transcripts_with_env):
         """
         Asynchronously evaluate all transcripts for an iteration.
 
         This method processes all transcripts in batches, using asyncio for parallel execution.
         It's specifically designed for use with the GPT backend.
 
         Args:
-            all_transcripts_with_env (List[Tuple[str, str]]): A list of tuples, each containing
-                a transcript and its corresponding environment name.
-            total_transcripts (int): The total number of transcripts to be evaluated.
+            all_transcripts_with_env (List[Tuple[int, Tuple[str, str]]]): A list of tuples, each containing
+                the index and a tuple of (transcript, env_name).
 
         Returns:
-            pd.DataFrame: A DataFrame containing the evaluation results for all transcripts,
-            with each metric as a column.
+            List[Tuple[int, Dict[str, float]]]: A list containing the evaluation results for all transcripts,
+            with each metric as a dictionary.
         """
         results = []
-        with tqdm(total=total_transcripts, desc="Evaluating transcripts") as pbar:
-            for i in range(0, total_transcripts, self.per_device_batch_size):
-                batch = all_transcripts_with_env[i : i + self.per_device_batch_size]
-                batch_results = self.evaluate_batch(batch, i, None)
-                results.append(batch_results)
+        backend = self.backend_class(
+            model_name=self.backend_config["model_name"],
+            model_id=self.backend_config["model_id"],
+            lora_path=self.backend_config["lora_path"],
+            device=None,
+        )
+        vectorized_assessors = self.vectorized_assessors_for_backend(backend)
+        with tqdm(total=len(all_transcripts_with_env), desc="Evaluating transcripts") as pbar:
+            batch_size = self.per_device_batch_size
+            i, end = 0, 0
+            while end < len(all_transcripts_with_env):
+                start = i * batch_size
+                end = start + batch_size
+                batch = all_transcripts_with_env[start:end]
```

Comment:
I'm confused as to why we're dealing in batches here? can't we pass 100% of the requests to openai backend at the same time and let it handle potential rate limits etc?